rustls/src/server/tls13.rs

```diff
@@ -197,79 +197,74 @@ mod client_hello {
                 });
             }
 
-            // choose a share that we support
-            let chosen_share_and_kxg = self
+            // Choose the most preferred common group.
+            let mutually_preferred_group = match self
                 .config
                 .provider
                 .kx_groups
                 .iter()
-                .find_map(|group| {
-                    shares_ext
-                        .iter()
-                        .find(|share| share.group == group.name())
-                        .map(|share| (share, *group))
-                });
+                .find(|group| groups_ext.contains(&group.name()))
+            {
+                Some(group) => *group,
+                None => {
+                    return Err(cx.common.send_fatal_alert(
+                        AlertDescription::HandshakeFailure,
+                        PeerIncompatible::NoKxGroupsInCommon,
+                    ));
```

Comment:
I was surprised there wasn't coverage here, but I think the reason is that we also detect this condition earlier, in `server/hs.rs` before calling tls13 `handle_client_hello`: https://github.com/rustls/rustls/blob/8e4afc6d141ba4e0df0f32e9079b2ac1c011d7b9/rustls/src/server/hs.rs#L364-L369

I'm not sure I spot a way we could get coverage here from an integration test as opposed to a more targeted unit test (and that doesn't seem worth the effort).

Reply:
Yeah I was looking at this, and it is unreachable now. There's a matching unreachable branch in tls12.rs. I am going to look into reorganising this so `NoKxGroupsInCommon` goes back to being raised when the kx group is chosen.

Reply:
Thanks, 8ef4b8d LGTM